src/core/path.c

```diff
@@ -144,6 +144,7 @@ void path_spec_unwatch(PathSpec *s) {
         assert(s);
 
         s->event_source = sd_event_source_unref(s->event_source);
+        inotify_rm_watch(s->inotify_fd,s->primary_wd);
```

Comment:
this appears unnecessary, we close the inotify ID right after, which which should remove all descriptors implicitly for us.

can you explain?

Reply:
Your statement  about closing file descriptor cleaning up of watches is correct. But what about if we see failures in closing? Kernel will still maintain the watch counts.

We are running systems in our arm based devices, 

The same code was working in 3.x kernel. But in 4.x kernels there are som issues around closing the file descriptors, due to which the watch count in the kernel space does not decrement.  Hence found reasonable to add code to inotify_rm_watch, which will reduce the counter, even if there are issues in closing the file descriptor.

Reply:
we generally try to avoid working around kernel bugs. If the watch count doesn't decrease even if the responsible inotify fd is fully closed, then that's a kernel bug, and really should be fixed there.